src/main/java/edu/ucsb/cs156/happiercows/controllers/JobsController.java

```diff
@@ -72,4 +72,30 @@ public Job launchMilkJob(
         return jobService.runAsJob(milkTheCowsJob);
     }
 
+    @ApiOperation(value = "Launch Update Cow Health Job (click fail if you want to test exception handling)")
+    @PreAuthorize("hasRole('ROLE_ADMIN')")
+    @PostMapping("/launch/updatehealthjob")
+    public Job launchUpdateHealthJob(
+    ) {
+
+        return jobService.runAsJob(ctx -> {
+            ctx.log("Starting to update cow health.");
+            ctx.log("This is where the code to update the cow's health will go");
+            ctx.log("Cows Health has been updated!");
+          });
+    }
+
+    @ApiOperation(value = "Launch Instructor Report Job (click fail if you want to test exception handling)")
+    @PreAuthorize("hasRole('ROLE_ADMIN')")
+    @PostMapping("/launch/instructorreportjob")
+    public Job launchInstructorReportJob(
+    ) {
+
+        return jobService.runAsJob(ctx -> {
+            ctx.log("Starting insrtuctor report");
+            ctx.log("This is where the code for the instructor report will go");
+            ctx.log("Instructor report job completed");
+          });
+    }
+
```

Comment:
Please take a look at the new code for the other jobs (TestJob, and MilkTheCowsJob), and restructure your code to match.  

You'll want to put the `ctx.log("...")` statements into their own classes under `jobs`, and then change the bodies of these methods to just launch the job.

Then the body of the two methods in this controller just becomes something like:

```java
        CowHealthUpdateJob cowHealthUpdateJob = CowHealthUpdateJob.builder().build();
        return jobService.runAsJob(cowHealthUpdateJob);
```

This is needed so that we can test these jobs effectively.

Reply:
Also: look at the tests, e.g. TestJobTests, MilkTheCowsJobTests, etc. as well as the tests in JobControllerTests.

Your tests in JobControllerTests will be similar to the one for Milking The Cow... you don't need to recreate all of the tests for TestJob.  If you have questions let me know!